x/mint/module/abci.go

```diff
@@ -217,8 +216,14 @@ func BeginBlocker(ctx context.Context, k keeper.Keeper) error {
 	}
 	if updateEmission {
 		// set the previous emissions to this block's emissions
-		k.PreviousRewardEmissionPerUnitStakedToken.Set(ctx, e_i)
-		k.PreviousBlockEmission.Set(ctx, blockEmission)
+		err = k.PreviousRewardEmissionPerUnitStakedToken.Set(ctx, e_i)
+		if err != nil {
+			return err
```

Comment:
ideally all errors wrapped in logger. You can see many other parts of code base as example

Reply:
Since I don't know the contexts yet, I did this first pass just to add the errors handling. But I will do another pass trying to add the contexts.